lib/statsd/instrument/udp_sink.rb

```diff
@@ -8,55 +11,74 @@ def self.for_addr(addr)
     new(host, Integer(port_as_string))
   end
 
+  class Dispatcher
+    def initialize(host, port, queue)
+      @host = host
+      @port = port
+      @queue = queue
+      @socket = nil
+      @thread = nil
+      ObjectSpace.define_finalizer(self, Proc.new { |o| o.stop })
+    end
+
+    def start
+      @thread = Thread.new { dispatcher_loop }
+      self
+    end
+
+    def stop
+      @queue.close
+      @thread.join(0.1)
+    rescue ThreadError
+    end
+
+    private
+    def dispatcher_loop
+      loop do
+        datagram = @queue.pop(false)
+        begin
+          socket.send(datagram, 0)
+        rescue ThreadError
+          socket.send(datagram, 0)
```

Comment:
How is `UDPSocket#send` expected to raise a `ThreadError` and why would you retry it?

Reply:
I'm assuming this was copypasta from https://github.com/Shopify/statsd-instrument/pull/257/files#diff-94686ddb5f351601e4da59e66916633fL29-L31. But I don't think it's necessary anymore now that there's no locking.